lessons/PowerShell/managed-identity-powershell.md

```diff
@@ -0,0 +1,240 @@
+# Get permission scope for "Group.Read.All"
+$appRoleId = az ad sp show --id $graphId --query "appRoles[?value=='Group.Read.All'].id | [0]" 
+```
+Time to make the REST call to assign the permissions as shown above to the Managed Identity:
+```
+#Set values
+$webAppName="LuiseDemo-functionapp$rand"
+$principalId=$(az resource list -n $webAppName --query [*].identity.principalId --out tsv)
+$graphResourceId=$(az ad sp list --display-name "Microsoft Graph" --query [0].objectId --out tsv)
+$appRoleId=$(az ad sp list --display-name "Microsoft Graph" --query "[0].appRoles[?value=='Group.Read.All' && contains(allowedMemberTypes, 'Application')].id" --out tsv)
+$body="{'principalId':'$principalId','resourceId':'$graphResourceId','appRoleId':'$appRoleId'}"
+
+#the actual REST call 
+az rest --method post --uri https://graph.microsoft.com/v1.0/servicePrincipals/$principalId/appRoleAssignments --body $body --headers Content-Type=application/json
+```
+
+> 📝 **Tip** - < you may control if everything worked as intended in Azure portal: Azure Active Directory --> Enterprise applications --> Managed Identity
+
+> 🔎 **Observation** - < OBSERVATION >
+
+> ❔ **Question** - < how would you several permissions in one go? >
+
+To see things work, we will need to deploy our function into our Functions App. 
+
+1. head over to Visual Studio Code again
+2. select **deploy to Functions App** 
+3. Select the Functions App yuou already created
+4. Confirm the Pop up window by selecting **Deploy**
+
+> 🔎 **Observation** - < this might take a minute >
+
+<!-- check with Marc if I need to explain how to test in Azure portal -->
+
+## 4. Obtain an access token
+
+In this step we want to learn how we could ontain an access token which we needed for a successful HTTP request against Microsoft Graph API: 
+
+### Steps
+
+1. Although we would usually load an authentication library such as Azure.Identity and then  to obtain a token, there is an easier, but not documented way get the token in an Azure  Functions: Following and extrapolating [Obtain tokens for Azure resources](https://docs.microsoft.com/en-us/azure/app-service/overview-managed-identity?tabs=powershell#obtain-tokens-for-azure-resources) to Microsoft Graph surprisingly works: 
+
+```
+$resourceURI = "https://<AAD-resource-URI-for-resource-to-obtain-token>"
+$tokenAuthURI = $env:IDENTITY_ENDPOINT + "?resource=$resourceURI&api-version=2019-08-01"
+$tokenResponse = Invoke-RestMethod -Method Get -Headers @{"X-IDENTITY-HEADER"="$env:IDENTITY_HEADER"} -Uri $tokenAuthURI
+$accessToken = $tokenResponse.access_token
+```
+
+2. have a look at your **IDENTITY_ENDPOINT** and **IDENTITY_HEADER** environment variables at `https://<your-functionappname-here>.scm.azurewebsites.net/ENV.cshtml#envVariables`
+
+## 5. Homework
+
+<!-- check with Marc what would be appropriate homework -->
+
+## 6. More info
+<!-- 
+check with Marc about what should go in this section -->
```

Comment:
Usually we put links here to MS docs. So in this case any links that have been used before in the text regarding background info on Managed Identities. E.g: https://github.com/marcduiker/azure-functions-university/blob/main/lessons/dotnetcore31/queue/queue-lesson-dotnet.md#9-more-info